tests/test_particle_sets.py

```diff
@@ -79,7 +149,18 @@ class TestParticle(ptype[mode]):
         p2 = Variable('p2', np.float32, initial=1, to_write=False)
         p3 = Variable('p3', np.float32, to_write='once')
 
-    pset = pset_type[pset_mode]['pset'](fieldset, lon=lon, lat=lat, depth=[4]*len(lon), pclass=TestParticle, p3=np.arange(len(lon)))
+    pset = None
+    if pset_mode != 'nodes':
+        pset = pset_type[pset_mode]['pset'](fieldset, lon=lon, lat=lat, depth=[4]*len(lon),
+                                            pclass=TestParticle, p3=np.arange(len(lon)))
+    else:
+        idgen = GenerateID_Service(SequentialIdGenerator)
+        idgen.setDepthLimits(0., 4.0)
+        idgen.setTimeLine(0.0, 4.0)
```

Comment:
comment in context to comment line 48: here we do change the parameters. We need to do that because the max depth is 4 and the time scale is 4 steps.